pkg/util/mount/mount_linux.go

```diff
@@ -926,19 +933,13 @@ func doCleanSubPath(mounter Interface, fullContainerDirPath, subPathIndex string
 	// process /var/lib/kubelet/pods/<uid>/volume-subpaths/<volume>/<container name>/<subPathName>
 	klog.V(4).Infof("Cleaning up subpath mounts for subpath %v", subPathIndex)
 	fullSubPath := filepath.Join(fullContainerDirPath, subPathIndex)
-	notMnt, err := IsNotMountPoint(mounter, fullSubPath)
-	if err != nil {
-		return fmt.Errorf("error checking %s for mount: %s", fullSubPath, err)
-	}
-	// Unmount it
-	if !notMnt {
-		if err = mounter.Unmount(fullSubPath); err != nil {
-			return fmt.Errorf("error unmounting %s: %s", fullSubPath, err)
-		}
-		klog.V(5).Infof("Unmounted %s", fullSubPath)
+
+	if err := UnmountMountPoint(fullSubPath, mounter, true); err != nil {
+		return fmt.Errorf("error unmounting %s: %s", fullSubPath, err)
 	}
+
 	// Remove it *non*-recursively, just in case there were some hiccups.
-	if err = os.Remove(fullSubPath); err != nil {
+	if err := os.Remove(fullSubPath); err != nil && !os.IsNotExist(err) {
```

Comment:
Is this necessary now that `UnmountMountPoint` is used which deletes the path?

Reply:
good catch, removed